src/protocols/secure_channel/CASESession.cpp

```diff
@@ -167,36 +167,34 @@ CHIP_ERROR CASESession::Deserialize(CASESessionSerialized & input)
 
 CHIP_ERROR CASESession::ToSerializable(CASESessionSerializable & serializable)
 {
-    // TODO: IPK should not be serialized/deserialized. Remove it from the stored session state.
     const NodeId peerNodeId = GetPeerNodeId();
     VerifyOrReturnError(CanCastTo<uint16_t>(mSharedSecret.Length()), CHIP_ERROR_INTERNAL);
     VerifyOrReturnError(CanCastTo<uint16_t>(sizeof(mMessageDigest)), CHIP_ERROR_INTERNAL);
-    VerifyOrReturnError(CanCastTo<uint16_t>(sizeof(mIPK)), CHIP_ERROR_INTERNAL);
     VerifyOrReturnError(CanCastTo<uint64_t>(peerNodeId), CHIP_ERROR_INTERNAL);
 
     memset(&serializable, 0, sizeof(serializable));
     serializable.mSharedSecretLen  = LittleEndian::HostSwap16(static_cast<uint16_t>(mSharedSecret.Length()));
     serializable.mMessageDigestLen = LittleEndian::HostSwap16(static_cast<uint16_t>(sizeof(mMessageDigest)));
-    serializable.mIPKLen           = LittleEndian::HostSwap16(static_cast<uint16_t>(sizeof(mIPK)));
-    serializable.mPairingComplete  = (mPairingComplete) ? 1 : 0;
     serializable.mVersion          = kCASESessionVersion;
     serializable.mPeerNodeId       = LittleEndian::HostSwap64(peerNodeId);
     serializable.mLocalSessionId   = LittleEndian::HostSwap16(GetLocalSessionId());
     serializable.mPeerSessionId    = LittleEndian::HostSwap16(GetPeerSessionId());
 
+    serializable.mCASESessionWasEstablished = (mCASESessionEstablished) ? 1 : 0;
```

Comment:
Isn't it just the case that we don't store an entry unless we successfully get a session established? If so, we wouldn't need this field.